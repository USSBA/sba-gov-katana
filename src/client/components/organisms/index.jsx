/* IMPORTS */

// miscellaenous organisms
import BusinessGuideTileCollection from "./business-guide-tile-collection/business-guide-tile-collection.jsx";
import CardCollection from "./card-collection/card-collection.jsx";
import DeveloperTester from "./developer-tester/developer-tester.jsx";
import DocumentCardCollection from "./document-card-collection/document-card-collection.jsx";
import FrontPageHero from "./front-page-hero/front-page-hero.jsx";
import FundingProgramsTileCollection from "./funding-programs-tile-collection/funding-programs-tile-collection.jsx";
import DisasterAlert from "./header-footer/disaster-alert/disaster-alert.jsx";
import NotificationBar from "./header-footer/notification-bar.jsx";
import Hero from "./hero/hero.jsx";
import RelatedDocumentCards from "./related-document-cards/related-document-cards.jsx";
import SearchBox from "./search-box/search-box.jsx";
import SectionNav from "./section-nav/section-nav.jsx";

// header-footer organisms
import Footer from "./header-footer/footer/footer.jsx";
import MobileNav from "./header-footer/mobile-nav/mobile-nav.jsx";
import Header from "./header-footer/header/header.jsx";

// homepage organisms
import Blog from "./homepage/blog/blog.jsx";
import FrontPageLady from "./homepage/front-page-lady/front-page-lady.jsx";
import HappeningNow from "./homepage/happening-now/happening-now.jsx";

// lender match organisms
// -- form
import AdditionalInfoForm from "./lender-match/form/additional-info.jsx";
import BusinessInfoForm from "./lender-match/form/business-info.jsx";
import ContactInfoForm from "./lender-match/form/contact-info.jsx";
import IndustryInfoForm from "./lender-match/form/industry-info.jsx";
import LoanInfo from "./lender-match/form/loan-info.jsx";
import ReviewSection from "./lender-match/form/review-page-helpers.jsx";
import ReviewSubmitInfoForm from "./lender-match/form/review-submit-info.jsx";
// -- landing page
import FindLendersIntro from "./lender-match/landing-page/find-lenders-intro.jsx";
import HelpfulQuestions from "./lender-match/landing-page/helpful-questions.jsx";
import {HowItWorksSection} from "./lender-match/landing-page/how-it-works.jsx";
import PreparationChecklist from "./lender-match/landing-page/preparation-checklist.jsx";
// -- success page
import ConfirmSection from "./lender-match/success-page/confirm-section.jsx";
import DynamicCounselingAndTools from "./lender-match/success-page/counseling-and-tools.jsx";
import CounselorMap from "./lender-match/success-page/counselor-map.jsx";
import EmailConfirmationInvalid from "./lender-match/success-page/email-confirmed-page.jsx";

// modal organisms
import LeaveSbaModal from "./modals/leave-sba-modal/leave-sba-modal.jsx";
import MobileSectionNavModal from "./modals/mobile-section-nav/mobile-section-nav.jsx";

/* EXPORTS */

// miscellaneous organisms
export {
	BusinessGuideTileCollection,
	CardCollection,
	DeveloperTester,
	DocumentCardCollection,
	FrontPageHero,
	FundingProgramsTileCollection,
	DisasterAlert,
	NotificationBar,
	Hero,
	RelatedDocumentCards,
	SearchBox,
	SectionNav
};

// header-footer organisms
export {
	Footer,
	MobileNav,
	Header
};

// homepage organisms
export {
	Blog,
	FrontPageLady,
	HappeningNow
};

// lender match organisms
export {
	AdditionalInfoForm,
	BusinessInfoForm,
	ContactInfoForm,
	IndustryInfoForm,
	LoanInfo,
	ReviewSection,
	ReviewSubmitInfoForm,
	FindLendersIntro,
	HelpfulQuestions,
	HowItWorksSection,
	PreparationChecklist,
	ConfirmSection,
	DynamicCounselingAndTools,
	CounselorMap,
	EmailConfirmationInvalid
};

// modal organisms
export {
	LeaveSbaModal,
	MobileSectionNavModal
};
