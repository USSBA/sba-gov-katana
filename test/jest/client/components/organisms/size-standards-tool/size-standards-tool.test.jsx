import React from "react";
/*eslint-enable no-unused-vars*/
import {SizeStandardsTool} from "organisms";
import {
	StartScreen,
	NaicsScreen,
	RevenueScreen,
	EmployeesScreen,
	ResultsScreen
} from "organisms/size-standards-tool/size-standards-tool.jsx";
import {shallow} from "enzyme";
import renderer from "react-test-renderer";

describe("SizeStandardsTool", () => {

	test("should match snapshot", () => {

		const tree = renderer.create(<SizeStandardsTool />).toJSON();
		expect(tree).toMatchSnapshot();

	});

	describe("StartScreen", () => {

		test("should match snapshot", () => {

			const tree = renderer.create(<StartScreen />).toJSON();
			expect(tree).toMatchSnapshot();

		});

	});

	describe("NaicsScreen", () => {

		const testProps = {
			naicsCodesList: [{
				"id": "111130",
				"description": "Dry Pea and Bean Farming",
				"sectorId": "11",
				"sectorDescription": "Agriculture, Forestry, Fishing and Hunting",
				"subsectorId": "111",
				"subsectorDescription": "Crop Production",
				"revenueLimit": null,
				"employeeCountLimit": 100,
				"footnote": null,
				"parent": null
			}]
		};

		test("should match snapshot", () => {

			const tree = renderer.create(<NaicsScreen {...testProps} />).toJSON();
			expect(tree).toMatchSnapshot();

		});

	});

	describe("RevenueScreen", () => {

		const mockFunction = jest.fn();

		const testProps = {
			setFocusTo: mockFunction
		};

		test("should match snapshot", () => {

			const tree = renderer.create(<RevenueScreen {...testProps} />).toJSON();
			expect(tree).toMatchSnapshot();

		});

	});

	describe("EmployeesScreen", () => {

		const mockFunction = jest.fn();

		const testProps = {
			setFocusTo: mockFunction
		};

		test("should match snapshot", () => {

			const tree = renderer.create(<EmployeesScreen {...testProps} />).toJSON();
			expect(tree).toMatchSnapshot();

		});

	});

	describe("ResultsScreen", () => {

		const testProps = {
			naicsCodesList: [{
				"id": "111130",
				"description": "Dry Pea and Bean Farming",
				"sectorId": "11",
				"sectorDescription": "Agriculture, Forestry, Fishing and Hunting",
				"subsectorId": "111",
				"subsectorDescription": "Crop Production",
				"revenueLimit": null,
				"employeeCountLimit": 100,
				"footnote": null,
				"parent": null,
				"isSmallBusiness": true
			}]
		};

		test("should match snapshot", () => {

			const tree = renderer.create(<ResultsScreen {...testProps} />).toJSON();
			expect(tree).toMatchSnapshot();

		});

	});

});