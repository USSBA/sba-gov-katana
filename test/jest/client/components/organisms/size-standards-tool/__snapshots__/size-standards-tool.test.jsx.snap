// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SizeStandardsTool EmployeesScreen should match snapshot 1`] = `
<div
  className="screen"
>
  <h2>
    How many employees?
  </h2>
  <div
    className="employeesInput"
  >
    <label
      htmlFor="employees"
    >
      Number of employees
    </label>
    <input
      id="employees"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onMouseUp={[Function]}
      type="text"
      value=""
    />
  </div>
  <p>
    This should be the average number of full-time or part-time 
    <br />
    employees over the last 12 months.
  </p>
  <button
    autoFocus={false}
    className="LargePrimaryButton  button"
    disabled={true}
    formTarget="_self"
    id={undefined}
    onKeyPress={[Function]}
    onTouchTap={[Function]}
    tabIndex="0"
  >
    SEE RESULTS
  </button>
</div>
`;

exports[`SizeStandardsTool NaicsScreen should match snapshot 1`] = `
<div
  className="screen"
>
  <h2>
    What's your industry?
  </h2>
  <div>
    <ul
      className="naicsCodesList"
    >
      <li>
        <div
          className="naicsSection"
        >
          <div
            className="left"
          >
            <p>
              <span>
                 
              </span>
            </p>
            <p>
              Dry Pea and Bean Farming
            </p>
          </div>
          <div
            className="right"
          >
            <a
              className="remove"
              onClick={[Function]}
              onKeyPress={[Function]}
              tabIndex="0"
            >
              <i
                aria-hidden="true"
                className="fa fa-times"
              />
            </a>
          </div>
        </div>
      </li>
    </ul>
  </div>
  <div>
    <div
      className="naicsCodeInput"
    >
      <div
        className="loading"
      >
        <p>
          ...loading suggestions...
        </p>
      </div>
    </div>
  </div>
  <div>
    <button
      autoFocus={false}
      className="LargePrimaryButton  button"
      disabled={undefined}
      formTarget="_self"
      id={undefined}
      onKeyPress={[Function]}
      onTouchTap={[Function]}
      tabIndex="0"
    >
      Next
    </button>
  </div>
</div>
`;

exports[`SizeStandardsTool ResultsScreen should match snapshot 1`] = `
<div
  className="screen"
>
  <h2>
    Are you a small business?
  </h2>
  <div>
    <div
      className="loading"
    >
      <p>
        ...now analyzing your business...
      </p>
    </div>
  </div>
  <div
    className="cards"
  >
    <div
      className="card"
    >
      <p>
        Learn more about 
        <a
          href="/contracting/getting-started-contractor/make-sure-you-meet-sba-size-standards"
          target="_blank"
        >
          SBA small business size standards
        </a>
        .
      </p>
      <p>
        <strong>
          SBA Office of Size Standards
        </strong>
      </p>
      <ul>
        <li>
          <i
            aria-hidden="true"
            className="fa fa-map-marker"
          />
          <p>
            409 3rd Street
            <span
              className="comma"
            >
              ,
            </span>
             SW 
            <br />
            Washington
            <span
              className="comma"
            >
              ,
            </span>
             DC 2041
          </p>
        </li>
        <li>
          <i
            aria-hidden="true"
            className="fa fa-phone"
          />
          <p>
            202-205-6618
          </p>
        </li>
        <li>
          <i
            aria-hidden="true"
            className="fa fa-envelope"
          />
          <p>
            <a
              href="mailto:sizestandards@sba.gov"
            >
              sizestandards@sba.gov
            </a>
          </p>
        </li>
      </ul>
    </div>
    <div
      className="card"
    >
      <p>
        Find out 
        <a
          href="/contracting"
          target="_blank"
        >
          how you can sell to the Federal Government
        </a>
        .
      </p>
      <p>
        <strong>
          SBA Office of Contracting
        </strong>
      </p>
      <ul>
        <li>
          <i
            aria-hidden="true"
            className="fa fa-map-marker"
          />
          <p>
            409 3rd Street
            <span
              className="comma"
            >
              ,
            </span>
             SW 
            <br />
            Washington
            <span
              className="comma"
            >
              ,
            </span>
             DC 2041
          </p>
        </li>
        <li>
          <i
            aria-hidden="true"
            className="fa fa-phone"
          />
          <p>
            202-205-6621
          </p>
        </li>
        <li>
          <i
            aria-hidden="true"
            className="fa fa-envelope"
          />
          <p>
            <a
              href="mailto:contracting@sba.gov"
            >
              contracting@sba.gov
            </a>
          </p>
        </li>
      </ul>
    </div>
  </div>
</div>
`;

exports[`SizeStandardsTool RevenueScreen should match snapshot 1`] = `
<div
  className="screen"
>
  <h2>
    How much revenue?
  </h2>
  <div
    className="revenueInput"
  >
    <label
      htmlFor="revenue"
    >
      Annual Revenue
    </label>
    <input
      id="revenue"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onMouseUp={[Function]}
      placeholder="$"
      type="text"
      value=""
    />
  </div>
  <p>
    This caption will help a small business understand 
    <br />
    what information we're looking for.
  </p>
  <button
    autoFocus={false}
    className="LargePrimaryButton  button"
    disabled={true}
    formTarget="_self"
    id={undefined}
    onKeyPress={[Function]}
    onTouchTap={[Function]}
    tabIndex="0"
  >
    SEE RESULTS
  </button>
</div>
`;

exports[`SizeStandardsTool StartScreen should match snapshot 1`] = `
<div>
  <h2>
    Size Standards Tool
  </h2>
  <img
    src="test-file-stub"
  />
  <p>
    Do you qualify as a small business?
  </p>
  <button
    autoFocus={false}
    className="LargePrimaryButton  button"
    disabled={undefined}
    formTarget="_self"
    id={undefined}
    onKeyPress={[Function]}
    onTouchTap={[Function]}
    tabIndex="0"
  >
    Start
  </button>
</div>
`;

exports[`SizeStandardsTool should match snapshot 1`] = `
<div
  className="sizeStandardsTool"
  id="size-standards-tool"
  tabIndex="-1"
>
  <div
    className="startSection"
  >
    <div>
      <h2>
        Size Standards Tool
      </h2>
      <img
        src="test-file-stub"
      />
      <p>
        Do you qualify as a small business?
      </p>
      <button
        autoFocus={false}
        className="LargePrimaryButton  button"
        disabled={undefined}
        formTarget="_self"
        id={undefined}
        onKeyPress={[Function]}
        onTouchTap={[Function]}
        tabIndex="0"
      >
        Start
      </button>
    </div>
  </div>
</div>
`;
